sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/Utility.java

```diff
@@ -482,4 +488,46 @@ public static SharedKeyCredential getSharedKeyCredential(HttpPipeline httpPipeli
         }
         return null;
     }
+
+    /**
+     * A utility method for converting the input stream to Flux of ByteBuffer. Will check the equality of
+     * entity length and the input length.
+     *
+     * @param data The input data which needs to convert to ByteBuffer.
+     * @param length The expected input data length.
+     * @param blockSize The size of each ByteBuffer.
+     * @return {@link ByteBuffer} which contains the input data.
+     * @throws UnexpectedLengthException when input data length mismatch input length.
+     * @throws RuntimeException When I/O error occurs.
+     */
+    public static Flux<ByteBuffer> convertStreamToByteBuffer(InputStream data, long length, int blockSize) {
```

Comment:
I feel like this is a common enough case that this should be in core? We can't be the only library moving data and need to do this conversion.

Reply:
Will discuss with @JonathanGiles if we can put up the helper method in Azure core. Leave it here right now as only storage has this kind of needs.

Reply:
Double check the Utility methods. There are bunch  a few which do not specific to storage. We can several of them into Azure core based on the needs of other clients. It is better to limit the scope of the utility to storage before preview 3.